pkg/kubectl/describe/versioned/describe.go

```diff
@@ -17,32 +17,4440 @@ limitations under the License.
 package versioned
 
 import (
+	"bytes"
+	"crypto/x509"
 	"fmt"
+	"io"
+	"net"
+	"net/url"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+	"text/tabwriter"
+	"time"
 
+	"github.com/fatih/camelcase"
+
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv2beta2 "k8s.io/api/autoscaling/v2beta2"
+	batchv1 "k8s.io/api/batch/v1"
+	batchv1beta1 "k8s.io/api/batch/v1beta1"
+	certificatesv1beta1 "k8s.io/api/certificates/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
+	networkingv1 "k8s.io/api/networking/v1"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	schedulingv1beta1 "k8s.io/api/scheduling/v1beta1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/duration"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+	externalclient "k8s.io/client-go/kubernetes"
+	coreclient "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	ref "k8s.io/client-go/tools/reference"
+	"k8s.io/klog"
 	"k8s.io/kubernetes/pkg/kubectl/describe"
-	"k8s.io/kubernetes/pkg/printers"
-	printersinternal "k8s.io/kubernetes/pkg/printers/internalversion"
+	"k8s.io/kubernetes/pkg/kubectl/scheme"
+	"k8s.io/kubernetes/pkg/kubectl/util/certificate"
+	deploymentutil "k8s.io/kubernetes/pkg/kubectl/util/deployment"
+	"k8s.io/kubernetes/pkg/kubectl/util/event"
+	"k8s.io/kubernetes/pkg/kubectl/util/fieldpath"
+	"k8s.io/kubernetes/pkg/kubectl/util/qos"
+	"k8s.io/kubernetes/pkg/kubectl/util/rbac"
+	resourcehelper "k8s.io/kubernetes/pkg/kubectl/util/resource"
+	"k8s.io/kubernetes/pkg/kubectl/util/slice"
+	storageutil "k8s.io/kubernetes/pkg/kubectl/util/storage"
+)
+
+// Each level has 2 spaces for PrefixWriter
+const (
+	LEVEL_0 = iota
+	LEVEL_1
+	LEVEL_2
+	LEVEL_3
+)
+
+const (
+	loadBalancerWidth = 16
+
+	// labelNodeRolePrefix is a label prefix for node roles
+	// It's copied over to here until it's merged in core: https://github.com/kubernetes/kubernetes/pull/39112
+	labelNodeRolePrefix = "node-role.kubernetes.io/"
+
+	// nodeLabelRole specifies the role of a node
+	nodeLabelRole = "kubernetes.io/role"
 )
 
 // DescriberFn gives a way to easily override the function for unit testing if needed
 var DescriberFn describe.DescriberFunc = Describer
```

Comment:
Move this to `pkg/kubectl/describe` instead of here. It reads weird and might cause cycles.

Reply:
This comment isn't still addressed.